File: Sources/UhSpotNavigation/UhSpotNavigationManager.m
//
//  CoreNavigationBridge.m
//  navigation
//
//  Created by User on 5/12/21.
//

#import <Foundation/Foundation.h>

#import "React/RCTBridgeModule.h"
#import "MapboxNavigation.h"
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(UhSpotNavigationManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(onLocationChange, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onRouteProgressChange, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onError, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onCancelNavigation, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onArrive, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(origin, NSArray)
RCT_EXPORT_VIEW_PROPERTY(destination, NSArray)
RCT_EXPORT_VIEW_PROPERTY(shouldSimulateRoute, BOOL)
RCT_EXPORT_VIEW_PROPERTY(showsEndOfRouteFeedback, BOOL)

@end

File: Sources/UhSpotNavigation/UhSpotNavigationManager.swift
//
//  corenavigation.swift
//  navigation
//
//  Created by User on 5/11/21.
//

import Foundation
import MapboxCoreNavigation

@objc(UhSpotNavigationManager)
class UhSpotNavigationManager: RCTViewManager {
  override func view() -> UIView! {
     return UhSpotNavigationView();
   }
  
  @objc func addEvent(name: String, location: String, date: NSNumber) -> Void {
    
  }
}
